### PR TITLE
update routerlicious to webpack 5

### PR DESCRIPTION
**Pull request by CraigMacomber**
# [Update to Webpack 5](#9568) (routerlicious part)

## Description

update routerlicious to webpack 5.

Add the deprecated [querystring npm package](https://www.npmjs.com/package/querystring) to package.json files for projects that import it: this was needed since a transitive dependency on it was removed, and breaking the preexisting but undeclared dependency.

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My code follows the code style of this project.
-   [ ] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.

## Testing

I ran the test suite for routerlicious. If there is a better way to test this I do not know what it is.

## Any relevant logs or outputs

> -   Use this section to provide either screenshots or output of logs as code snippets

## Other information or known dependencies

> -   Any other information or known dependencies that is important to this PR.
> -   TODO that are to be done after this PR.

### CODE REVIEW

**Comment by CraigMacomber** [comment]
Looks like our repo has 5 usages of the deprecated [querystring npm package](https://www.npmjs.com/package/querystring), but never declares it as a dependency in any of the package.json files. This means our code relies on it being brought in by a transitive dependency, and it appears that updating webpack caused it to not be rough in transitively anymore for at least one case.

**Comment by CraigMacomber** [comment]
CI is reporting a failing test with a rather interesting and reverent error, but I cannot reproduce this error locally: the test just passes for me, despite doing a full git clean.

**Comment by CraigMacomber** [comment]
I have explicitly opted out of buffer and util polyfills in server-local-server and server-lambdas. I'm unsure if anything consumes the webpack bundles for those, and I don't know how to test them. Maybe they wok, maybe they will error due to missing functionality if used?

File: server/routerlicious/packages/local-server/webpack.config.js [hunk 1]
```diff
@@ -23,7 +23,12 @@ module.exports = {
         ]
     },
     resolve: {
-        extensions: [ '.tsx', '.ts', '.js' ]
+        extensions: [ '.tsx', '.ts', '.js' ],
+        fallback: {
+            // To polyfill buffer, use: require.resolve("buffer/")
```
Comment (anthony-murphy) [inline server/routerlicious/packages/local-server/webpack.config.js:28]: if you do symlink:full do the examples' tests pass? Many have jest tests that will launch chrome with local server and valid the example works

Comment (CraigMacomber) [reply server/routerlicious/packages/local-server/webpack.config.js:28]: As far as I can tell, the bundle this produces it not used (its not the entry point to the package, and its not imported by the entry point, not is it imported directly by the users of this module either). Those examples are already on webpack 5, have separate configs, and I tested their polyfill settings by running their tests and running them interactively when I updated them to webpack 5.

Anyway, with symlink full I ran `npx lerna run --scope "@fluid-example/*" test:jest --concurrency=1` which passed, which I think answers your question.

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**